fdbclient/DatabaseBackupAgent.actor.cpp

```diff
@@ -1296,19 +1296,25 @@ namespace dbBackup {
 					}
 
 					if (backupRanges.size() == 1) {
-						state Key destUidLookupPath = BinaryWriter::toValue(backupRanges[0], IncludeVersion()).withPrefix(destUidLookupPrefix);
-						Optional<Key> existingDestUidValue = wait(srcTr->get(destUidLookupPath));
-						if (existingDestUidValue.present()) {
-							if (destUidValue == existingDestUidValue.get()) {
-								// due to unknown commit result
-								break;
-							} else {
-								// existing backup/DR is running
-								return Void();
+						Standalone<RangeResultRef> existingDestUidValues = wait(srcTr->getRange(KeyRangeRef(destUidLookupPrefix, strinc(destUidLookupPrefix)), CLIENT_KNOBS->TOO_MANY));
+						bool found = false;
+						for(auto it : existingDestUidValues) {
+							if( BinaryReader::fromStringRef<KeyRange>(it.key.removePrefix(destUidLookupPrefix), IncludeVersion()) == backupRanges[0] ) {
+								if(destUidValue != it.value) {
+									// existing backup/DR is running
+									return Void();
```

Comment:
Before this would return if any results came back from the read, now it returns only if the results are for the same range. I haven't fully thought through the upgrade case, but I wanted to make sure that's a correct change of behavior.

Reply:
The key only contains a range and the FDB version. The previous code would only match if both the range was the same and the version was the same. The new code matches if just the range is the same, even if the FDB version is different, which is the intended behavior.